### PR TITLE
issue #454 ToolInstallation fails to load every other restart

### PR DESCRIPTION
**Pull request by vektory79**
Introduced pocessing of overrided property setters with type concretization

### CODE REVIEW

**Comment by casz** [comment]
Can you please rebase to resolve conflict?

File: plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java [hunk 1]
```diff
@@ -24,12 +24,7 @@
 import java.lang.reflect.Type;
 import java.lang.reflect.TypeVariable;
 import java.lang.reflect.WildcardType;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: Please avoid star imports

**Comment by vektory79** [comment]
Most of time I use Mercurial. So looks like I do something wrong with this PR. I can recreate PR if anybody suggest it. :-(

**Comment by ndeloof** [comment]
@vektory79 I cherry picked your fix, thanks for working on this issue !

**Pull request closed without merging**